testdata/sh/update-kernel-deps.sh

```diff
@@ -30,14 +30,25 @@ cleanup() {
 
 trap cleanup EXIT
 
-# Download and process libbpf.c
-curl -fL "https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/plain/tools/lib/bpf/libbpf.c?h=v$KERNEL_VERSION" -o "$tmp/libbpf.c"
+if [ -d $KERNEL_VERSION ]; then
+  # Copy libbpf/vmlinux/bpf_testmod from local directory
+  cp $KERNEL_VERSION/libbpf.c $tmp
```

Comment:
sorry I meant to create draft ;-) I'm including this branch in tetragon to enable and use kprobe.session programs
and then I wanted to do the official  6.11 interface update if you wouldn't beat me to it

I'm using the that KERNEL_VERSION change to use my kernel with changed interface for testing
and I usually do not have the kernel tree on the same machine so I just copy those files in the
directory and feed it to update-kernel-deps.sh script

I'm not clear on how you guys add the new ci kernel though.. but I can find out ;-)

Reply:
> I'm using the that KERNEL_VERSION change to use my kernel with changed interface for testing and I usually do not have the kernel tree on the same machine so I just copy those files in the directory and feed it to update-kernel-deps.sh script

I see. In that case, would you be willing to adjust your workflow to mount the tree locally, e.g. using sshfs and running the script against the mount? I think most folks would iterate using virtme locally, so this could still be useful for them.

> I'm not clear on how you guys add the new ci kernel though.. but I can find out ;-)

Images are built and published by https://github.com/cilium/ci-kernels. rc's are pulled from Linus' tree, mainline is pulled from kernel.org (see https://github.com/cilium/ci-kernels/blob/main/download.sh). No bpf-next currently, maybe we can consider this at some point. (so you'll have to wait for an rc at least :smile:)